test/quicer_SUITE.erl

```diff
@@ -423,6 +426,40 @@ tc_peername_v4(Config) ->
       ct:fail("listener_timeout")
   end.
 
+tc_alpn(Config) ->
+  Port = 4573,
+  Owner = self(),
+  Opts = lists:keyreplace(alpn, 1, default_listen_opts(Config), {alpn, ["sample2", "sample"]}),
+  {SPid, _Ref} = spawn_monitor(fun() -> conn_server_with(Owner, Port, Opts) end),
+  receive
+    listener_ready ->
+      {ok, Conn} = quicer:connect("localhost", Port, default_conn_opts(), 5000),
+      {ok, {_, _}} = quicer:sockname(Conn),
+      ok = quicer:close_connection(Conn),
+      SPid ! done
+  after 1000 ->
+    ct:fail("timeout")
+  end.
+
+tc_alpn_mismatch(Config) ->
+  Port = 4574,
+  Owner = self(),
+  Opts = lists:keyreplace(alpn, 1, default_listen_opts(Config), {alpn, ["no"]}),
+  {SPid, _Ref} = spawn_monitor(fun() -> conn_server_with(Owner, Port, Opts) end),
+  receive
+    listener_ready ->
+      spawn_monitor(fun() -> quicer:connect("localhost", Port, default_conn_opts(), 5000),
```

Comment:
We haven't done the code path for the closed connection (peer close).

I will fix it when the code is ready.